mitodl/open-edx-plugins
# feat: Add Rapid Response reports plugin

### Related Tickets:
https://github.com/mitodl/open-edx-plugins/issues/9, https://github.com/mitodl/open-edx-plugins/issues/18 & https://github.com/mitodl/edx-platform/issues/273

### What's this PR does?
- Adds rapid response reports plugins to support reports creation for `Rapid response xBlock`


### How to test?
To test this PR you will need a counterpart in the `edx-platform` and that part is specifically done in https://github.com/mitodl/edx-platform/pull/275

- Setup your edx-platform on the associated branch https://github.com/mitodl/edx-platform/pull/275 from here
- Install this plugin (e.g. generate a build for this plugin with `./pants package ::` (You might need to first build the pants itself using `./pants build` before creating package) and then install that into the edx-platform)
- You also need to install the rapid response xblock (https://github.com/mitodl/rapid-response-xblock) in order for the plugin to work
- Restart the lms (`make lms-restart`)
- You should create some multiple choice problems with `Rapid response plugin` enabled (See the instructions on the Rapid Response xBlock https://github.com/mitodl/rapid-response-xblock)
- Now open a course as admin and click `Instructor` tab and make sure you see a `Rapid Responses` tab under it
- Click the `Rapid Responses` tab and test all the functionality of Rapid Respons reports downloads
     - You should see lists of timestamped links and clicking any link should download a CSV formatted report with user/answer details

### Background context:
We had made several changes inside our for of the edx-platform where we added a Django application to support rapid response reports. In the process, we made many changes in open edX's `Instructor` Django application since we wanted to run `Rapid Reports reports` inside the `Instrutor` tab. 

### Screenshots:
<img width="1025" alt="Screenshot 2021-11-19 at 5 38 12 PM" src="https://user-images.githubusercontent.com/34372316/142624210-34827bff-65ac-4ca8-92c2-071cb402e46b.png">
